pkg/generate/app/cmd/newapp.go

```diff
@@ -662,7 +662,7 @@ func (c *AppConfig) buildPipelines(components app.ComponentReferences, environme
 			switch {
 			case refInput.ExpectToBuild:
 				glog.V(4).Infof("will add %q secrets into a build for a source build of %q", strings.Join(c.Secrets, ","), refInput.Uses)
-				if err := refInput.Uses.AddBuildSecrets(c.Secrets); err != nil {
+				if err := refInput.Uses.AddBuildSecrets(c.Strategy, c.Secrets); err != nil {
```

Comment:
@mfojtik sorry for the late response... so the flag we set on the source repo is the most reliable. I would change the strategy parameter to a bool isDockerBuild and pass refInput.Uses.IsDockerBuild() for the value.